enterprise/api/filters.py

```diff
@@ -54,3 +54,24 @@ def filter_queryset(self, request, queryset, view):
             queryset = queryset.filter(user_id=request.user.id)
 
         return queryset
+
+
+class EnterpriseCustomerFilterBackend(filters.BaseFilterBackend):
```

Comment:
Would a more generic filter name make sense here? At the end of the day, this filter is for filtering out non-linked EnterpriseCustomerUsers, so perhaps the name should reflect that. That way, the filter could more easily be used for other viewsets as well where relevant.

Maybe `EnterpriseLinkedUserFilterBackend` or similar?